flocker/control/test/test_script.py

```diff
@@ -114,12 +133,15 @@ def test_starts_cluster_state_service(self):
         ``ControlScript.main`` starts a cluster state service.
         """
         options = ControlOptions()
-        options.parseOptions(
-            [b"--port", b"tcp:8001", b"--data-path", self.mktemp()])
+        options.parseOptions([
+            b"--port", b"tcp:8001", b"--data-path", self.mktemp(),
+            b"--certificates-directory", self.certificate_path.path
+        ])
         reactor = MemoryCoreReactor()
         ControlScript().main(reactor, options)
         server = reactor.tcpServers[0]
-        service = server[1].resource._v1_user.cluster_state_service
+        control_resource = server[1].wrappedFactory.resource
+        service = control_resource._v1_user.cluster_state_service
```

Comment:
I think Demeter gets angry even if you spread your `.`s across multiple statements.

Going through `wrappedFactory` seems particularly egregious and may be what tips this over the edge from bad to worth-fixing (not that `resource` is a lot better).  The `_v1_user` trick seems like the lesser evil here.  How about an attribute referring to `cluster_state_service` (since that's what we actually want to get) instead?  Then this is **only** `server[1].resource._cluster_state_service`.